src/raven.js

```diff
@@ -1950,6 +1951,48 @@ function hasKey(object, key) {
   return objectPrototype.hasOwnProperty.call(object, key);
 }
 
+function matchFilters(filters, object) {
+  var args = Array.prototype.slice.call(arguments, 1),
+    i = 0,
+    len = filters.length,
+    filter;
+  for (; i < len; i++) {
+    filter = filters[i];
+    if (isFunction(filter) && filter.apply(null, args)) {
+      return true;
+    } else if (isRegExp(filter) && filter.test(object)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+function optimizeFilters(filters) {
+  // Optimize filters array by joining strings and regular expressions
+  // and pass along function predicate.
+  var regexps = [],
+    optimized_filters = [],
+    i = 0,
+    len = filters.length,
```

Comment:
JS engines already optimise it themselves, so there's no need to store the length :)